test/new-e2e/tests/containers/docker/docker_test.go

```diff
@@ -0,0 +1,141 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package docker for all kubernetes new E2E tests
+package docker
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/containers"
+	dockervm "github.com/DataDog/test-infra-definitions/scenarios/aws/dockerVM"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/stretchr/testify/suite"
+)
+
+type DockerSuite struct {
+	containers.BaseSuite
+	isDsdStandalone bool
+	stackName       string
+}
+
+func TestDockerSuite(t *testing.T) {
+	// Full Agent
+	suite.Run(t, &DockerSuite{stackName: "docker-stack", isDsdStandalone: false})
+
+	// DogstatsD Standalone
+	suite.Run(t, &DockerSuite{stackName: "docker-stack", isDsdStandalone: true})
+}
+
+func (suite *DockerSuite) getAgentImage() string {
+	if suite.isDsdStandalone {
+		return "datadog/dogstatsd:latest"
+	}
+
+	return ""
+}
+
+func (suite *DockerSuite) SetupSuite() {
+
+	ctx := context.Background()
+
+	stackConfig := runner.ConfigMap{
+		"ddagent:deploy":        auto.ConfigValue{Value: "true"},
+		"ddagent:fakeintake":    auto.ConfigValue{Value: "true"},
+		"ddagent:fullImagePath": auto.ConfigValue{Value: suite.getAgentImage()},
+	}
+
+	if suite.isDsdStandalone {
+		stackConfig["ddagent:fullImagePath"] = auto.ConfigValue{Value: "datadog/dogstatsd:latest"}
+	}
+
+	_, stackOutput, err := infra.GetStackManager().GetStackNoDeleteOnFailure(ctx, suite.stackName, stackConfig, dockervm.Run, false)
+
+	if !suite.Assert().NoError(err) {
+		suite.FailNow(ctx)
+	}
+
+	vmConnection := stackOutput.Outputs["vm-connection"].Value.(map[string]interface{})
+	suite.stackName = fmt.Sprintf("%s-%v", os.Getenv("USER"), vmConnection["host"])
+
+	fakeintakeHost := stackOutput.Outputs["fakeintake-host"].Value
+	suite.Fakeintake = fakeintake.NewClient(fmt.Sprintf("http://%s", fakeintakeHost))
+
+	suite.Eventuallyf(
+		func() bool { return suite.Fakeintake.GetServerHealth() == nil },
+		1*time.Minute,
+		5*time.Second,
+		"Fakeintake is never healthy",
+	)
+
+	err = suite.Fakeintake.FlushServerAndResetAggregators()
+	if !suite.Assert().NoError(err) {
+		fmt.Println("Failed to flush/reset fakeintake")
+		suite.FailNow(ctx)
+	}
+
+	suite.BaseSuite.SetupSuite()
+}
+
+func (suite *DockerSuite) FailNow(ctx context.Context) {
+	_, err := infra.GetStackManager().GetPulumiStackName(suite.stackName)
+	suite.Require().NoError(err)
+	if !runner.GetProfile().AllowDevMode() || !*containers.KeepStacks {
+		infra.GetStackManager().DeleteStack(ctx, suite.stackName)
+	}
+	suite.T().FailNow()
```

Comment:
Would it make sense to move this logic to the test framework? It feels like it should apply to all tests, not just the docker ones.

Reply:
I assume you mean having this method defined on the `baseSuite` so it is shared by all E2E tests. (Correct if I am wrong)

Yes I certainly agree, but I prefer keeping this to a separate PR because it will require making simple changes to all existing tests. I prefer to keep this PR focused on DSD E2E.